src/internal/lokiutil/testloki/load.go

```diff
@@ -0,0 +1,105 @@
+package testloki
+
+import (
+	"bufio"
+	"context"
+	"io"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+)
+
+func AddLogFile(ctx context.Context, r io.Reader, l *TestLoki) error {
+	s := bufio.NewScanner(r)
+	labels := map[string]string{}
+	var i int
+	var logs []*Log
+	for s.Scan() {
+		i++
+		line := s.Text()
+		if line == "" {
+			continue
+		}
+		switch {
+		case strings.HasPrefix(line, "&map[") && strings.HasSuffix(line, "]"):
+			labels = parseLabels(line)
+		default:
+			log := parseLog(i, line)
+			if log.Time.IsZero() {
+				return errors.Errorf("line %d (%q): no time", i, line)
+			}
+			log.Labels = make(map[string]string)
+			for k, v := range labels {
+				log.Labels[k] = v
+			}
+			logs = append(logs, log)
+		}
+	}
+	if err := s.Err(); err != nil {
+		return errors.Wrap(err, "scan")
+	}
+	if err := l.AddLog(ctx, logs...); err != nil {
+		return errors.Wrapf(err, "line %d: AddLog", i)
+	}
+	return nil
+}
+
+// parseLabels parses a &map[key:value key2:value2] line.  These appear in the loki-logs.txt files
+// from a debug dump.  The line must have the prefix "&map[" and suffix "]".
+func parseLabels(line string) map[string]string {
+	result := map[string]string{}
+	var key, value strings.Builder
+	var state bool // false = accumulate key, true = accumulate value
+	for _, c := range line[len("&map[") : len(line)-1] {
+		switch {
+		case c == ' ':
```

Comment:
This breaks on maps with spaces in either the keys or the values.

The default Go printed representation of a map is inherently ambiguous: `map[string]string{"foo:bar": "baz"}` and `map[string]string{"foo": "bar:baz"}` both print as `map[foo:bar:baz]`. It gets worse! `map[foo:bar baz foo bar:baz]` might have originally been:

- `map[string]string{"foo":"bar", "baz foo bar":"baz"}`
- `map[string]string{"foo":"bar baz", "foo bar":"baz"}` 
- `map[string]string{"foo":"bar baz foo"}`
- `map[string]string{"foo":"bar baz foo bar:baz"}`
- `map[string]string{"foo:bar baz foo bar":"baz"}`
- `map[string]string{"foo":"", "bar baz foo bar":"baz"}`

Given that we control the test files, this is not such a big deal, but given that we control the test files, we could also specify an unambiguous format. Worth documenting if you choose to stick with the default Go map format.

Reply:
It's not an ideal format, but it's what we're already using in the debug server.  &map is also very unlikely to appear in logs, so it's a good signal that this is our metadata.  (But the original goal of the format was to be human-readable, not parseable.  I don't think these can be parsed unambiguously: 

```go
package main

import "fmt"

func main() {
        fmt.Printf("%v\n", map[string]string{"foo:bar bar": "baz"})
}
```
```
map[foo:bar bar:baz]
```
So it goes.